File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/jwt/entity/BlackListToken.java
package com.luckyseven.backend.sharedkernel.jwt.entity;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class BlackListToken {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String tokenValue;

  private Instant expirationTime;

  @Builder
  public BlackListToken(String tokenValue, Instant expirationTime) {
    this.tokenValue = tokenValue;
    this.expirationTime = expirationTime;
  }


}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/jwt/entity/BlackListToken.kt
package com.luckyseven.backend.sharedkernel.jwt.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "black_list_token", indexes = [
   Index(name = "idx_blacklist_token_value", columnList = "tokenValue"),
   Index(name = "idx_blacklist_expiration_time", columnList = "expirationTime")
])
 class BlackListToken(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    
    val tokenValue: String,
    
    val expirationTime: Instant
) 